VisualPinball.Unity/VisualPinball.Unity.Editor/Managers/Switches/SwitchListViewItemRenderer.cs

```diff
@@ -0,0 +1,348 @@
+								{
+									selectedIndex = tmpIndex;
+								}
+
+								tmpIndex++;
+							}
+						}
+
+						EditorGUI.BeginChangeCheck();
+						int index = EditorGUI.Popup(cellRect, selectedIndex, options.ToArray());
+						if (EditorGUI.EndChangeCheck())
+						{
+							switchListData.InputActionMap = inputSystemList[index].ActionMapName;
+							switchListData.InputAction = inputSystemList[index].ActionName;
+							updateAction(switchListData);
+						}
+					}
+					break;
+				case SwitchSource.Playfield:
+					{
+						List<string> options = new List<string>();
+						foreach (var item in _switchables)
+						{
+							options.Add(item.Name);
+						}
+
+						EditorGUI.BeginChangeCheck();
+						int index = EditorGUI.Popup(cellRect, options.IndexOf(switchListData.PlayfieldItem), options.ToArray());
+						if (EditorGUI.EndChangeCheck())
+						{
+							switchListData.PlayfieldItem = options[index];
+							updateAction(switchListData);
+						}
+					}
+					break;
+				case SwitchSource.Constant:
+					{
+						EditorGUI.BeginChangeCheck();
+						int index = EditorGUI.Popup(cellRect, (int)switchListData.Constant, OPTIONS_SWITCH_CONSTANT);
+						if (EditorGUI.EndChangeCheck())
+						{
+							switchListData.Constant = index;
+							updateAction(switchListData);
+						}
+					}
+					break;
+			}
+		}
+
+		private void RenderType(SwitchListData switchListData, Rect cellRect, Action<SwitchListData> updateAction)
+		{
+			if (switchListData.Source == SwitchSource.InputSystem || switchListData.Source == SwitchSource.Playfield)
+			{
+				EditorGUI.BeginChangeCheck();
+				int index = EditorGUI.Popup(cellRect, (int)switchListData.Type, OPTIONS_SWITCH_TYPE);
+				if (EditorGUI.EndChangeCheck())
+				{
+					switchListData.Type = index;
+					updateAction(switchListData);
+				}
+			}
+		}
+
+		private void RenderOff(SwitchListData switchListData, Rect cellRect, Action<SwitchListData> updateAction)
+		{
+			if (switchListData.Source == SwitchSource.InputSystem || switchListData.Source == SwitchSource.Playfield)
+			{
+				if (switchListData.Type == SwitchType.Pulse)
+				{
+					var labelRect = cellRect;
+					labelRect.x += labelRect.width - 20;
+					labelRect.width = 20;
+
+					var intFieldRect = cellRect;
+					intFieldRect.width -= 25;
+
+					EditorGUI.BeginChangeCheck();
+					var pulse = EditorGUI.IntField(intFieldRect, switchListData.Pulse);
+					if (EditorGUI.EndChangeCheck())
+					{
+						switchListData.Pulse = pulse;
+						updateAction(switchListData);
+					}
+
+					EditorGUI.LabelField(labelRect, "ms");
+				}
+			}
+		}
+
+		private UnityEngine.Texture GetIcon(SwitchListData switchListData)
+		{
+			string image = null;
+
+			if (switchListData.Source == SwitchSource.Playfield)
+			{
+				foreach (var item in _switchables)
+				{
+					if (item.Name == switchListData.PlayfieldItem)
+					{
+						if (item is BumperAuthoring)

```

Comment:
Maybe a Dictionary<System.Type, string> lookup should work there.
it would reduce the image setup to one line (excluding the dictionary declaration but it would be more compact).

Reply:
Great, I will give that try. I actually was thinking about that all along. `is` was twisting me up.

Reply:
the only thing is for managing a default case (maybe with a try...catch), but there you had only know types.
I think you have to use typeof() in dictionary entries and the item.GetType() for accessing, well i tried on something else and it seems to work.

Reply:
@jsm174 proposed to move it into the Authoring class directly, which I like even better.